scripts/lint.sh

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+
+# Treat undefined variables and non-zero exits in pipes as errors.
+set -uo pipefail
+
+# Ensure that the "**" glob operator is applied recursively.
+shopt -s globstar
+
+# Break on first error.
+set -e
+
+# Parse input arguments.
+diff_only=0
+for i in "$@"; do
+    case $i in
+        --diff )
+            diff_only=1
+            shift;;
+
+        *)
+            echo "Unknown option: $i";
+            exit 1;;
+    esac
+done
+
+cd "$(git rev-parse --show-toplevel)"
```

Comment:
Do the `git -C` and `dirname` trick here that will allow running this script from any directory. See the other scripts in the `scripts` directory.